launch_handler-explainer.md

```diff
@@ -0,0 +1,141 @@
+# Web App Launch Handling
+
+## Overview
+
+This document describes a new `launch_handler` manifest member that enables
+web apps to customise their launch behaviour across all types of app launch
+triggers.
+
+
+## Use Cases
```

Comment:
Updated overview to explain on declarative approach.

launch_handler is a subset of DLC, they may co-exist if the overlap is removed from DLC (though sw-launch would probably not the right place for it anymore).
This part is out of scope for launch_handler:
> *   Link capturing for PWAs: a PWA that wants to open in a window whenever the user clicks a link to a URL within that app’s [navigation scope](https://www.w3.org/TR/appmanifest/#navigation-scope), rather than opening in a browser tab.
> *   Capturing links and navigations from the following (non-exhaustive list of) sources:
>     *   Clicked links from other web pages.
>     *   URL launch from a native app in the operating system.
Added that to a non-goals section.

Mentioned DLC in the background section.

Security & privacy isn't the same without link capturing in scope, that section still looks DLC specific.

Added missing related proposal sections.